### DISCUSSION TITLE
How to use FontAwesome icons in Menubar?

### DISCUSSION THREAD

**Discussion by agarny** [primefaces/community#3252] [PrimeVue] [answered]
Hi,

I have integrated [FontAwesome icons](https://fontawesome.com/icons) in my Vue.js app as described in https://docs.fontawesome.com/web/use-with/vue and it's all working fine. IOW, if I have something like `<FontAwesomeIcon icon="house" />` then I can see https://fontawesome.com/icons/house?f=classic&s=solid in my app, as expected.

But what if I want to use FontAwesome icons in some PrimeVue components such as [Menubar](https://primevue.org/menubar/)? Say that I have `<Menubar :model="items" />` with `items` equal to:
```javascript
const emit = defineEmits(['about'])

const items = ref([
  {
    label: 'Help',
    items: [
      {
        label: 'About',
        icon: 'XXX', // <--- What do I put here?...
        command: () => {
          emit('about')
        }
      }
    ]
  }
])
```
Then, what do I set `icon` to? Hopefully, I am missing something very obvious...?

Cheers, Alan.

**Comment by AutoAwesome** [comment, marked as answer]
I had the same issue.

To use font-awesome-icons in your menu start the dom.watcher of font-awesome in your main.js `dom.watch()`. Then you can use ie. `fa-solid fa-link` as `icon`.

**Comment by agarny** [comment]
Well, I am clearly missing something since I am not able to get it work:
- I have the following in my `package.json` file:
```json
"@fortawesome/fontawesome-svg-core": "^6.7.2",
"@fortawesome/free-solid-svg-icons": "^6.7.2",
"@fortawesome/vue-fontawesome": "^3.0.8",
```
- I also have the following my main file:
```typescript
import * as fa from '@fortawesome/fontawesome-svg-core'
import { fas } from '@fortawesome/free-solid-svg-icons'
...

fa.dom.watch()
fa.library.add(fas)
...
```

Then, I tried:
```typescript
{
  label: 'About',
  icon: 'home',
  command: () => {
    emit('about')
  }
}
```

but the only visual difference is that there is now a space before `About` in my menu. As previously mentioned, the following works fine:
```typescript
<FontAwesomeIcon icon="home" />
```

Anyway, although I would like to use icons in a `MenuBar` object, I would also like to use icons in my Electron menu (yes, my app is both a Web app _per se_ and an Electron app) and I feel like the latter might not be possible...?

**Comment by AutoAwesome** [comment]
I've created an example for you. You can see here in this menubar how to use a PI-Icon and a FA-Icon:
https://stackblitz.com/edit/goca6gte?file=src%2FApp.vue

**Comment by agarny** [comment]
Thanks for this @AutoAwesome! Your original message was actually enough. My mistake was to assume that, like in my use of `FontAwesomeIcon`, I could simply specify `home` rather than `fa-solid fa-home`. IOW, I am all good using the following:
Then, I tried:
```typescript
{
  label: 'About',
  icon: 'fa-solid fa-home', // <-- Indeed, what you said in your first message...
  command: () => {
    emit('about')
  }
}
```
To have `icon: 'fa-solid home'` is not good enough either. For menu items, we must absolutely prepend with `fa-` (as well as specify the icon family, i.e. `fa-solid`). Ok, lesson learned!
